File: greetings-service/greetings-service-domain/src/test/java/net/bakaar/greetings/domain/glue/GreetingsCreationSteps.java
package net.bakaar.greetings.domain.glue;

import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;
import net.bakaar.greetings.domain.Greeting;
import net.bakaar.greetings.domain.exception.GreetingWrongTypeException;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.catchThrowable;


public class GreetingsCreationSteps {

    private Greeting greeting;
    private Throwable thrown;

    @When("I create a(n) {word} greeting for {word}")
    public void iCreateAGreetingForName(String type, String name) {
        thrown = catchThrowable(() -> greeting = Greeting.of(type).to(name).build());
    }

    @Then("I get the message {string}")
    public void iGetTheMessage(String message) {
        assertThat(greeting.getMessage()).isEqualTo(message);
    }

    @Then("I get an error")
    public void iGetAnError() {
        assertThat(thrown).isNotNull().isInstanceOf(GreetingWrongTypeException.class);
    }
}

File: greeting-ui-e2e/cypress/e2e/step_definitions/GreetingsCreation.steps.ts
import {Then, When} from "@badeball/cypress-cucumber-preprocessor";

When('I create a(n) {word} greeting for {word}', (type: string, name: string) => {
    cy.createGreeting(type, name);
});


Then('I get the message {string}', (message: string) => {
    cy.verifyGreetingMessage(message);
});

Then('a Greeting is created', () => {
    cy.verifyGreetingCreated();
});

Then('I get an error', () => {
    cy.verifyErrorMessage();
});